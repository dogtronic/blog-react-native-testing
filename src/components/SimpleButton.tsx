import React from 'react';
import styled from 'styled-components/native';

type ButtonProps = {
  label: string;
  onPress: () => void;
  buttonTestID?: string;
};

const ButtonContainer = styled.TouchableOpacity`
  height: ${56}px;
  width: 100%;
  margin-top: 17px;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  background-color: ${props => props.theme.colors.background.dark};
`;

const ButtonText = styled.Text`
font-family: "RobotoMono-Medium"
font-size:16px
color:
 ${props => props.theme.colors.text.lightGrey}
line-height:21px
`;

const SimpleButton: React.FC<ButtonProps> = ({
  label,
  onPress,
  buttonTestID,
}) => {
  return (
    <ButtonContainer onPress={onPress} testID={buttonTestID}>
      <ButtonText>{label}</ButtonText>
    </ButtonContainer>
  );
};

export default SimpleButton;
