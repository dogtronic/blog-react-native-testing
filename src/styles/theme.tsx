import React, {ReactElement} from 'react';
import * as styledComponents from 'styled-components/native';

type colors = {
  system: {
    tokenOnWhite: string;
    tokenGreen: string;
    tokenGreen60: string;
    tokenGreen30: string;
    tokenGreen10: string;
    tabBarBorder: string;
  };
  text: {
    black: string;
    darkGrey: string;
    grey: string;
    lightGrey: string;
    highlightGrey: string;
    white: string;
  };
  status: {
    dangerous: string;
    bad: string;
    good: string;
  };
  background: {
    light: string;
    dark: string;
  };
};

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: colors;
  }
}

export const theme: styledComponents.DefaultTheme = {
  colors: {
    system: {
      tokenOnWhite: '#14CA85',
      tokenGreen: '#00E69D',
      tokenGreen60: '#6FFFE3',
      tokenGreen30: '#B9FFE8',
      tokenGreen10: '#E7FFF7',
      tabBarBorder: '#E2E3E4',
    },
    text: {
      black: '#000000',
      darkGrey: '#36354A',
      grey: '#686777',
      lightGrey: '#B3B5BD',
      highlightGrey: '#D2D2D7',
      white: '#FFFFFF',
    },
    status: {
      dangerous: '#FF0000',
      bad: '#FF6B00',
      good: '#15AA2C',
    },
    background: {
      light: '#FFFFFF',
      dark: '#F8F8FA',
    },
  },
};

export type ThemeInterface = typeof theme;

const {
  default: styled,
  css,
  ThemeProvider,
} = styledComponents as styledComponents.ReactNativeThemedStyledComponentsModule<ThemeInterface>;

const withTheme = (element: ReactElement) => {
  <ThemeProvider theme={theme}>{element}</ThemeProvider>;
};

export {styled, css, ThemeProvider, withTheme};
