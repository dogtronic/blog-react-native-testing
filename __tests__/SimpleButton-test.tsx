import SimpleButton from '../src/components/SimpleButton';
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'styled-components/native';
import {theme} from '../src/styles/theme';

describe('SimpleButton', () => {
  test('renders correctly', () => {
    const simpleButtonRender = renderer
      .create(
        <ThemeProvider theme={theme}>
          <SimpleButton onPress={jest.fn()} label={'Test'} />
        </ThemeProvider>,
      )
      .toJSON();

    expect(simpleButtonRender).toMatchSnapshot();
  });
});
