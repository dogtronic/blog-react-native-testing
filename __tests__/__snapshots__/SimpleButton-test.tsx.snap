// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleButton renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F8F8FA",
      "flexDirection": "row",
      "height": 56,
      "justifyContent": "center",
      "marginTop": 17,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#B3B5BD",
          "fontFamily": "RobotoMono-Medium",
          "fontSize": 16,
          "lineHeight": 21,
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;
